Update external-oauth.md (#6433)
fishtown-analytics/docs.getdbt.com: website/docs/faqs/Troubleshooting/failed-snowflake-oauth-connection.md

---

> title: Receiving a `Failed to connect to DB xxxxxxx.snowflakecomputing.com:443` error when when connecting to Snowflake via external Oauth

Author: ```suggestion
title: Receiving a `Failed to connect to DB` error when connecting to Snowflake
```

description: "Edit your OAuth Security integration when you see error"
sidebar_label: 'Receiving `Failed to connect to database` error'
---

1. If you see this error: 

> ```

Author: ```suggestion
```shell
```

> If you see the following error w

Author: ```suggestion
1. If you see this error: 
```

Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. Please specify another role, or contact your OAuth Authorization server administrator.
```

> To resolve this issue, you should edit your OAuth Security integration and explicitly specify this Scope mapping attribute:

Author: ```suggestion
2. Edit your OAuth Security integration and explicitly specify this Scope mapping attribute:
```

 ```sql
ALTER INTEGRATION <my_int_name> SET EXTERNAL_OAUTH_SCOPE_MAPPING_ATTRIBUTE = 'scp';
```

> You can read more about this error in [Snowflake's documentation](https://community.snowflake.com/s/article/external-custom-oauth-error-the-role-requested-in-the-connection-is-not-listed-in-the-access-token).

Author: ```suggestion

You can read more about this error in [Snowflake's documentation](https://community.snowflake.com/s/article/external-custom-oauth-error-the-role-requested-in-the-connection-is-not-listed-in-the-access-token).
```

---
title: Receiving a `Failed to connect to DB` error when connecting to Snowflake
description: "Edit your OAuth Security integration when you see error"
sidebar_label: 'Receiving `Failed to connect to database` error'
---

1. If you see this error: 

```shell

> Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. Please specify another role, or contact your OAuth Authorization server administrator.

Author: ```suggestion
Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. 
Please specify another role, or contact your OAuth Authorization server administrator.
```

---
title: Receiving a `Failed to connect to DB` error when connecting to Snowflake
description: "Edit your OAuth Security integration when you see error"
sidebar_label: 'Receiving `Failed to connect to database` error'
---

1. If you see this error: 

```shell

> Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. 

Author: ```suggestion

Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. 
```

Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. 

> Please specify another role, or contact your OAuth Authorization server administrator.

Author: ```suggestion
Please specify another role, or contact your OAuth Authorization server administrator.

```
```

---
title: Receiving a `Failed to connect to DB` error when connecting to Snowflake
description: "Edit your OAuth Security integration when you see error"
sidebar_label: 'Receiving `Failed to connect to database` error'
---

1. If you see this error: 

```shell

Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. 
Please specify another role, or contact your OAuth Authorization server administrator.

> ```

Author: ```suggestion

```
```

---
title: Receiving a `Failed to connect to DB` error when connecting to Snowflake
description: "Edit your OAuth Security integration when you see error"
sidebar_label: 'Receiving `Failed to connect to database` error'
---

1. If you see this error: 

   ```shell
   

>    Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection or the default role if none was requested in the connection ('xxxxx') is not listed in the Access Token or was filtered. 

Reviewer: ```suggestion
   Failed to connect to DB: xxxxxxx.snowflakecomputing.com:443. The role requested in the connection, or the default role if none was requested in the connection ('xxxxx'), is not listed in the Access Token or was filtered. 
```

   Please specify another role, or contact your OAuth Authorization server administrator.
   
   ```

> 2. Edit your OAuth Security integration and explicitly specify this Scope mapping attribute:

Reviewer: ```suggestion
2. Edit your OAuth Security integration and explicitly specify this scope mapping attribute:
```

Reviewer: Hiya @runleonarun 

I've checked the snowflake docs and I'm not 100% sure if the "s" in Scope mapping attribute should be capitalised or not. In some of the Snowflake docs, the "s" is capitalised and in others, it isn't.

Kind Regards
Natalie
